primary/src/grpc_server/validator.rs

```diff
@@ -2,40 +2,89 @@
 // SPDX-License-Identifier: Apache-2.0
 use std::time::Duration;
 
-use crate::{block_waiter::GetBlockResponse, BlockCommand};
+use crate::{block_waiter::GetBlockResponse, BlockCommand, BlockRemoverCommand};
 use tokio::{
-    sync::{mpsc::Sender, oneshot},
+    sync::{mpsc::channel, mpsc::Sender, oneshot},
     time::timeout,
 };
 use tonic::{Request, Response, Status};
 use types::{
-    BatchMessageProto, BlockError, CollectionRetrievalResult, GetCollectionsRequest,
+    BatchMessageProto, BlockError, CollectionRetrievalResult, Collections, Empty,
     GetCollectionsResponse, Validator,
 };
 
 #[derive(Debug)]
 pub struct NarwhalValidator {
     tx_get_block_commands: Sender<BlockCommand>,
     get_collections_timeout: Duration,
+    tx_block_removal_commands: Sender<BlockRemoverCommand>,
+    remove_collections_timeout: Duration,
 }
 
 impl NarwhalValidator {
     pub fn new(
         tx_get_block_commands: Sender<BlockCommand>,
         get_collections_timeout: Duration,
+        tx_block_removal_commands: Sender<BlockRemoverCommand>,
+        remove_collections_timeout: Duration,
     ) -> Self {
         Self {
             tx_get_block_commands,
             get_collections_timeout,
+            tx_block_removal_commands,
+            remove_collections_timeout,
         }
     }
 }
 
 #[tonic::async_trait]
 impl Validator for NarwhalValidator {
+    async fn remove_collections(
+        &self,
+        request: Request<Collections>,
+    ) -> Result<Response<Empty>, Status> {
+        let collection_ids = request.into_inner().collection_ids;
+        let remove_collections_response = if !collection_ids.is_empty() {
+            let (tx_remove_block, mut rx_remove_block) = channel(1);
+            let mut ids = vec![];
+            for collection_id in collection_ids {
+                ids.push(collection_id.try_into().map_err(|err| {
+                    Status::invalid_argument(format!("Could not serialize: {:?}", err))
+                })?);
+            }
```

Comment:
Since we are performing the exact same conversions for [get_collections](https://github.com/MystenLabs/narwhal/pull/224/files#diff-0a5c9d58935199886da37e8caaccb463c22ffc45caff1340e467f819f0616d7fR92) , maybe we could introduce a method to reuse in both places?